internal/pproflogging/pproflogging.go

```diff
@@ -237,30 +284,31 @@ func StartProfileBuffers(ctx context.Context) {
 }
 
 // DumpPem dump a PEM version of the byte slice, bs, into writer, wrt.
-func DumpPem(bs []byte, types string, wrt *os.File) error {
+func DumpPem(ctx context.Context, bs []byte, types string, wrt Writer) error {
 	// err0 for background process
 	var err0 error
 
 	blk := &pem.Block{
 		Type:  types,
 		Bytes: bs,
 	}
+
 	// wrt is likely a line oriented writer, so writing individual lines
 	// will make best use of output buffer and help prevent overflows or
 	// stalls in the output path.
 	pr, pw := io.Pipe()
+
 	// encode PEM in the background and output in a line oriented
 	// fashion - this prevents the need for a large buffer to hold
 	// the encoded PEM.
 	go func() {
-		// writer close on exit of background process
-		//nolint:errcheck
-		defer pw.Close()
 		// do the encoding
 		err0 = pem.Encode(pw, blk)
-		if err0 != nil {
-			return
-		}
+
```

Comment:
Be a bit better at cleaning up resources... PipeWriters/PipeReaders don't return meaningful errors on close ... so just ignore them (plus less code for delta code coverage to complain about)